### ISSUE TITLE
Leveraging uncertainty information from deep neural networks for disease detection

### ISSUE THREAD

**Issue by agitter** [greenelab/deep-review#122]
http://doi.org/10.1101/084210

> In recent years, deep neural networks (DNNs) have revolutionized the field of computer vision and image processing. In medical imaging, algorithmic solutions based on DNNs have been shown to achieve high performance on tasks that previously required medical experts. So far DNN-based solutions for disease detection have been proposed without quantifying their uncertainty in a decision. In contrast, a physician knows whether she is uncertain about a case and will consult more experienced colleagues if needed. Here we propose to estimate the uncertainty of DNNs in medical diagnosis based on a recent theoretical insight on the link between dropout networks and approximate Bayesian inference. Using the example of detecting diabetic retinopathy (DR) from fundus photographs, we show that uncertainty informed decision referral improves diagnostic performance. Experiments across different networks, tasks and datasets showed robust generalization. Depending on network capacity and task/dataset difficulty, we surpass 85% sensitivity and 80% specificity as recommended by the NHS when referring 0%-20% of the most uncertain decisions for further inspection. We analyse causes of uncertainty by relating intuitions from 2D visualizations to the high-dimensional image space, showing that it is in particular the difficult decisions that the networks consider uncertain.

Using an approximate Bayesian model to identify uncertain predictions that should be manually assessed would be good to highlight in the review.  The data primarily come from Kaggle: "35,126 training images and 53,576 test images, graded into five stages of [diabetic retinopathy]" with another 1,200 images from an independent source.
